_rules/inline-link-in-paragraph-is-distinguishable-be4d0c.md

```diff
@@ -33,11 +33,13 @@ acknowledgements:
 
 This rule applies to any HTML element which:
 
-- has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
-- has [descendant][] [text nodes][text node] in the [flat tree][] that are [visible][]; and
-- is a [descendant][] in the [flat tree][] of an HTML element (with the closest [ancestor][] in the [flat tree][] being henceforth referred to as the _ancestor_ element) that has a [computed value][] for the [display property][display] that is not one of the following: 'table', 'flex', 'grid', 'ruby', 'table-row-group', 'table-header-group', 'table-footer-group', 'table-row', table-column-group', 'table-column', 'ruby-base-container' and 'ruby-text-container'; and
-- the _ancestor_ element has other [descendant][] [visible][] [text nodes][text node] in the [flat tree][]; and
-- the [descendant][] [visible][] [text nodes][text node] of the _ancestor_ element belong to the same [inline formatting context][].
+- **(is a link)** has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
+- **(visible)** has [descendant][] [text nodes][text node] in the [flat tree][] that are [visible][]; and
+- **(belongs to a block container)** is a [descendant][] in the [flat tree][] of an HTML element (with the closest [ancestor][] in the [flat tree][] being henceforth referred to as the _ancestor_ element) that has a [computed value][] for the [display property][display] that is not one of the following: 'table', 'flex', 'grid', 'ruby', 'table-row-group', 'table-header-group', 'table-footer-group', 'table-row', table-column-group', 'table-column', 'ruby-base-container' and 'ruby-text-container'; and
```

Comment:
For me, this is the one that is more difficult to understand. I am not really sure how we could clarify this @WilcoFiers  @carlosapaduarte . 

Just trying to help clarifying this. as said above.



```suggestion
- **(belongs to a block container)** is a [descendant][] in the [flat tree][] of an HTML element whose closest [ancestor][] (the _ancestor_ element) has a [computed value][] for its [display property][display] that is not one of the following: 'table', 'flex', 'grid', 'ruby', 'table-row-group', 'table-header-group', 'table-footer-group', 'table-row', table-column-group', 'table-column', 'ruby-base-container' and 'ruby-text-container'; and
```

* Is there a need to mention the flat tree twice? By specifying it at the beginning of the sentence, I think it is clear that we are consistently meaning "the flat tree". If we were to mean another tree in the same sentence, we could well specify it.
* We are using "closest ancestor". Do we really want to mean "grand-parent node?

Reply:
The multiple mentions to the flat tree are there to ensure the applicability is not ambiguous. 

The closest ancestor is the closest ancestor. It might be the parent in the DOM, or the great-parent, or the great-great-parent. It depends on what nodes of the DOM are in the flat tree.